sklearn/ensemble/gradient_boosting.py

```diff
@@ -1367,9 +1381,12 @@ def staged_predict_proba(self, X):
         try:
             for score in self.staged_decision_function(X):
                 yield self.loss_._score_to_proba(score)
-        except AttributeError:
-            raise AttributeError('loss=%r does not support predict_proba' %
-                                 self.loss)
+        except AttributeError as e:
+            if isinstance(e, NotFittedError):
+                raise e
+            else:
+                raise AttributeError('loss=%r does not support predict_proba' %
+                                     self.loss)
```

Comment:
On 01/05/2015 04:33 PM, ragv wrote:
>
> In sklearn/ensemble/gradient_boosting.py 
> <https://github.com/scikit-learn/scikit-learn/pull/4029#discussion-diff-22491360>:
>
> > @@ -1367,9 +1381,12 @@ def staged_predict_proba(self, X):
> >          try:
> >              for score in self.staged_decision_function(X):
> >                  yield self.loss_._score_to_proba(score)
> > -        except AttributeError:
> > -            raise AttributeError('loss=%r does not support predict_proba' %
> > -                                 self.loss)
> > +        except AttributeError as e:
> > +            if isinstance(e, NotFittedError):
> > +                raise e
> > +            else:
> > +                raise AttributeError('loss=%r does not support predict_proba' %
> > +                                     self.loss)
>
> Oh..! Thanks for pointing this out... That was a typo sorry! There 
> should have been an |e| at 1358...
>
>
Did travis not catch this?